File: pgm/ABC_Java/ABC_460_A.java
import java.io.*;
// import java.math.*;
// import java.time.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        reader.close();
        final int one = 1;
        final int zero = 0;
        final int N = Integer.parseInt (arr[zero]);
        int M = Integer.parseInt (arr[one]);
        int cnt = zero;
        while (M > zero) {
            cnt++;
            M = N % M;
        }
        System.out.println (cnt);
        System.exit (0);
    }
}

File: pgm/ABC_C#/ABC_460_A.cs
using System;
// using System.Collections.Generic;
// using System.Globalization;
// using System.IO;
// using System.Linq;
// using System.Text;
// using System.Text.RegularExpressions

class Program
{
    static void Main()
    {
        string[] arr = Console.ReadLine().Split();
        const int one = 1;
        const int zero = 0;
        int N = int.Parse(arr[zero]);
        int M = int.Parse(arr[one]);
        int cnt = zero;
        while (M > zero) {
            cnt++;
            M = N % M;
        }
        Console.WriteLine(cnt);
        Environment.Exit(0);
    }
}